src/note/nullifier.rs

```diff
@@ -63,21 +63,16 @@ impl Nullifier {
 pub mod testing {
     use proptest::prelude::*;
 
-    use group::GroupEncoding;
-    use pasta_curves::pallas;
-
     use super::Nullifier;
-    use crate::spec::extract_p;
+    use rand::{rngs::StdRng, SeedableRng};
 
     prop_compose! {
         /// Generate a uniformly distributed nullifier value.
```

Comment:
Hmm, I think I'm going to go back on my prior feedback on this method. Instead of making the output match the docs, I think we should make the method output a nullifier in the full `pallas::Base` range that the specification allows for (even though that is outside the range of a real nullifier). The `proptest` backend is going to be applying its own strategies to the input, so we aren't actually aiming for "uniformly distributed", but "can reach the full space".
```suggestion
        /// Generate a nullifier within the `pallas::Base` domain.
        ///
        /// This can return nullifiers that fall outside the range of valid nullifiers
        /// (which are in the space of valid `pallas::Point` x-coordinates).
```